src/main/java/org/protege/editor/owl/server/http/handlers/HTTPChangeService.java

```diff
@@ -33,6 +33,36 @@
 	private final ServerLayer serverLayer;
 	private final ChangeService changeService;
 
+	static enum CanCommitSquash {
+		OK, NOT_WORKFLOW_MANAGER, NOT_PAUSING_USER;
+
+		String message() {
+			switch (this) {
+				case OK:
+					return "";
+				case NOT_WORKFLOW_MANAGER:
+					return "you aren't a workflow manager";
+				case NOT_PAUSING_USER:
+					return "Server in maintenance mode, please try later";
+			}
+			return "impossible"; // impossible to reach
+		}
+
+		boolean ok() {
+			return this.equals(OK);
+		}
+
+		static CanCommitSquash create(ProjectId projectId, User user) {
+			if (!HTTPServer.server().isWorkFlowManager(user, projectId)) {
+				return NOT_WORKFLOW_MANAGER;
+			} else if (!HTTPServer.server().isPausingUser(user)) {
+				return NOT_PAUSING_USER;
+			} else {
+				return OK;
+			}
+		}
+	}
+
```

Comment:
Not sure I buy this premature refactoring. This enum incorporates two pieces of state that are used together in one place along one application path.